File: android/app/src/main/kotlin/com/tbtrapp/data/ContactsCache.kt
package com.tbtrapp.data

import androidx.compose.runtime.mutableStateOf

/**
 * App-wide contacts cache — same pattern as GlobalAudioPlayer.
 *
 * Because this is a plain Kotlin object (not screen-scoped remember state),
 * it survives navigating away from and back to ContactScreen. The list is
 * fetched once per app process (first time the screen is ever opened), and
 * every re-entry after that reads instantly from here — no spinner, no
 * re-query. A quiet background sync (see ContactScreen) then checks for
 * newly-registered matches without blocking the UI.
 */
object ContactsCache {
    val contacts = mutableStateOf<List<AppContact>>(emptyList())
    var hasLoadedOnce = false
}

File: lib/data/contacts_cache.dart
import 'package:flutter/foundation.dart';
import 'app_contact.dart';

/// Equivalent of ContactsCache — app-wide singleton (like GlobalAudioPlayer,
/// 6c), not screen-scoped, so leaving and returning to the Contacts screen
/// reads instantly from here rather than re-querying every time.
class ContactsCache extends ChangeNotifier {
  ContactsCache._internal();
  static final ContactsCache instance = ContactsCache._internal();

  List<AppContact> _contacts = [];
  List<AppContact> get contacts => _contacts;

  bool hasLoadedOnce = false;

  void setContacts(List<AppContact> value) {
    _contacts = value;
    notifyListeners();
  }
}